src/pybind/mgr/dashboard/controllers/multi_cluster.py

```diff
@@ -87,18 +79,66 @@ def auth(self, url: str, name: str, username=None, password=None, token=None, hu
                     component='dashboard')
 
             token = content['token']
-            # Set CORS endpoint on remote cluster
+
+        if token:
             self._proxy('PUT', url, 'ui-api/multi-cluster/set_cors_endpoint',
-                        payload={'url': multicluster_config['hub_url']}, token=token)
+                            payload={'url': hub_url}, token=token)
+            fsid = self._proxy('GET', url, 'api/health/get_cluster_fsid', token=token)
 
             multicluster_config['config'].append({
-                'name': name,
-                'url': url,
-                'token': token
-            })
+                    'name': fsid,
+                    'cluster_alias': cluster_alias,
+                    'url': url,
+                    'token': token
+                })
+        Settings.MULTICLUSTER_CONFIG = multicluster_config
+
+    @Endpoint('PUT')
+    @UpdatePermission
+    def set_config(self, config: str):
+        if isinstance(Settings.MULTICLUSTER_CONFIG, str):
+            item_to_dict = json.loads(Settings.MULTICLUSTER_CONFIG)
```

Comment:
same thing i pointed above. if not a proper json, this will fail. you need to properly handle the exceptions